packages/flutter_tools/templates/package_ffi/pubspec.yaml.tmpl

```diff
@@ -0,0 +1,19 @@
+name: {{projectName}}
+description: {{description}}
+version: 0.0.1
+homepage:
+
+environment:
+  sdk: {{dartSdkVersionBounds}}
+
+dependencies:
+  cli_config: ^0.1.1
```

Comment:
Do we want to actively push a tool-level dependency into the package ecosystem? That seems like the wrong support level for something being used this way in our own template.

Reply:
> I believe we should graduate that package once we decide to take the native assets feature out of experimental.

If we haven't actually decided to take it out of experimental, then it really doesn't seem like we should be building templates for it into `flutter` that will encourage people to build and publish packages using it. If we don't actually have a decision, why not just publish a template as a one-off repository to point people to, instead of building the tooling into `flutter` first?

Reply:
We need support in `flutter run`/`flutter assemble`. I guess we could omit `flutter create` support.

Would it be possible to gate `flutter create` support behind the experimental flag as well?

Reply:
That seems reasonable.

Reply:
If the discussion here is about the criteria for graduating package:cli_config from tools.dart.dev to dart.dev, that's just gated on us believing the functionality is general purpose enough, and the current owner+team agreeing to maintaining the package at a high level SLO (https://pub.dev/publishers/dart.dev/packages vs https://pub.dev/publishers/tools.dart.dev/packages).

Reply:
> We need support in `flutter run`/`flutter assemble`. I guess we could omit `flutter create` support.
> 
> Would it be possible to gate `flutter create` support behind the experimental flag as well?

This SGTM

Reply:
I have gated the template behind the feature.

> us believing the functionality is general purpose enough

I believe so. We wouldn't want every `build.dart` dev reinvent the wheel on:

1. How to parse the yaml config and output the yaml output in the communication with the Dart and Flutter SDK.
2. Reinvent the wheel on how to invoke a native C toolchain from a Dart script.

However, I believe package:native_toolchain_c can be made much more feature rich than it is now.

> the current owner+team agreeing to maintaining the package at a high level SLO

Packages: cli_config and native_assets_cli we (the native interop team) will maintain with high SLO. I expect most contributions to come from ourselves. These packages are the counterpart of this PR and the CLs in the Dart SDK in the protocol between the SDKs and the packages.

For package native_toolchain_c, I expect that to be more on the reviewers side as users start demanding more features from it and submit PRs. We will shepherd these contributions.